pkg/logging/grpc.go

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package logging
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors"
+	grpc_logging "github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/tags"
+	"github.com/oklog/ulid"
+	"google.golang.org/grpc/status"
+	"gopkg.in/yaml.v2"
+)
+
+// NewReqLogConfig parses the string into a req logging config structure.
+// Raise an error if unmarshalling is not possible, or values are not valid.
+func NewReqLogConfig(configYAML []byte) (*ReqLogConfig, error) {
```

Comment:
Here, we could be explicit 
```suggestion
func NewRequestConfig(configYAML []byte) (*RequestConfig, error) {
```

Reply:
Sounds great